src/main/java/com/redhat/devtools/intellij/tektoncd/utils/TreeHelper.java

```diff
@@ -96,9 +99,14 @@ public static Tree getTree(Project project) {
         } else if (node instanceof TaskRunNode) {
             content = tkncli.getTaskRunYAML(namespace, node.getName());
             kind = KIND_TASKRUN;
+            readonly = true;
+        } else if (node instanceof PipelineRunNode){
+            content = tkncli.getPipelineRunYAML(namespace, node.getName());
+            kind = KIND_PIPELINERUN;
+            readonly = true;
         }
 
-        return Pair.pair(content, kind);
+        return new Triple<>(content, kind, readonly);
```

Comment:
It is difficult for me to comment here (all the code related to Pair/Triple) as I'm not a seasoned java developer so maybe @jeffmaury can shed some light. Coming from c# where i used `structs` and javascript with `objects` this makes completely sense to me (i was the one who used Pair at first i think) but i'm not sure if in java this is considered good code or not as it can be confusing to know which values (first, second, third) represent. I have the fear that by keeping adding stuff to this, the triple could become a four-value tuple in future, then five-values and so on... @jeffmaury WDYT? How should one behave in such situation?

Looking at the code, i would have kept it as it was and just check the kind inside `createAndOpenVirtualFile` and if it was a `taskrun` or `pipelinerun` set the writable to false. This way you could have changed less code i guess. 

BTW don't keep this comment in consideration as this is just my written thought to understand better what to do in such cases. Better to wait the review by Jeff

Reply:
Unfortunately java do not have  `structs` or easy way to return new 'object' class. 
I'm also to use TS a lot, and for me it is strange to create a new class, just for return multiple values from function/method. In java common way to solve this, it use dedicated class for result. And I think if I create an use class witch will describe Tekton resource which we want to open will simplify `createAndOpenVirtualFile`, we may reduce number of params in it.

> For `Pair` and `Triple` I was looking on some presentation from JetBrains guy, and he discuss why they have this class in IT platform, he say that it is comfort way to return multiple value from function/method, without introducing additional type, which will be used as container for values. But if you want to return more that 3 values than you need to have own type. But, of course, it is matter of code guidelines that we decide to apply on this case.